src/services/ArmorsServices.ts

```diff
@@ -38,7 +46,16 @@ export default class ArmorsServices extends ValidateEntry implements Service<Int
     }
 
     public async update(_id: string, payload: Internacional<Armor>): Promise<Internacional<Armor>> {
-        this.validate(languagesWrapper(armorZodSchema), payload);
+        this.validate(languagesWrapper(armorsZodSchema), payload);
+
+        if (payload.active) {
+            const err = new Error('Not authorized to change availability');
```

Comment:
Se o erro realmente for `400 - Bad Request,` seria interessante uma mensagem equivalente:
```suggestion
            const err = new Error('Could not change availability');
```

A mensagem atual encaixaria melhor num `401 - Unauthorized`.
É mais uma questão de semântica, o que acha @Adson-Gomes-Oliveira?

Reply:
o 401 é relacionado à alguem que não tem autorização para realizar determinada ação, o usuário tem autorização para mudar o availability porém não faz sentido ele mudar para o mesmo status nem que os valores da query param sejam diferentes de true ou false.

Então o erro realmente é 400 uma vez que ele tem autorização para realizar a operação, porém está realizando de maneira errada. ( No caso está tentando mudar para ativo algo que já está ativo ).

Reply:
>Então o erro realmente é 400

Então mantém a frase atual?

Reply:
Eu mudaria para
`Entity availability is already active` ou `Entity availability is already disabled`

Mas a forma que o @ashtrindade apontou também é válida, fica ao teu critério @RasecMH

Reply:
É que nesse caso @Adson-Gomes-Oliveira é quando o usuário tentar mudar a disponibilidade pela rota de update, a mensagem de que já está habilitado ou desabilitado faz sentido na rota onde existe essa opção que é a de updateAvailability. 

Talvez alterar para "Not authorized to change availability via this route." para deixar mais claro faça sentido, o que vocês acham?

Reply:
@RasecMH @ashtrindade 

Hmmmm perfeito, tinha entendido errado, nesse caso realmente, mudar a disponibilidade atráves da rota de update não é permitido, porém, o código correto para essa situação é o 403, forbidden, uma vez que, mudar a disponibilidade pela rota de update seria possível, mas não é permitido NEM para administradores ou algo do tipo, logo:

- Não é um problema de autorização pois ninguém deve utilizar essa rota para mudar disponibilidade
- Não é um problema de bad request, pois a rota de update teoricamente pode mudar qualquer coisa e faz sentido poder mudar a disponibilidade, nada impede.

- Porém é um problema de regra de negocio, mesmo que nada impeça em nível de lógica o active ser mudado na rota de update, foi decidido que essa operação, apesar de possível não é interessante que possa ser feita através da rota de update por questões de segurança, logo a operação de mudar a disponibilidade na rota de update independe de nível de autorização, é possivel de ser feita na rota de update, porém por decisões internas essa operação ( através da rota de update ) é proibida. Justificando o uso do código de erro 403.

Sugestão de mensagem: `Change availability through update route is forbidden`